avocado/utils/process.py

```diff
@@ -1349,3 +1349,39 @@ def system_output(cmd, timeout=None, verbose=True, ignore_status=False,
     if strip_trail_nl:
         return cmd_result.stdout.rstrip(b'\n\r')
     return cmd_result.stdout
+
+
+def getoutput(cmd):
+    """
+    Becauce commands module is removed in Python3 and it redirect stderr
+    to stdout, we port commands.getoutput to make code compatible
+    Return output (stdout or stderr) of executing cmd in a shell.
+
+    :param cmd: Command line to run.
+    :type cmd: str
+
+    :return: Command output(stdout or stderr).
+    :rtype: str
+    """
+    return getstatusoutput(cmd)[1]
+
+
+def getstatusoutput(cmd):
+    """
+    Becauce commands module is removed in Python3 and it redirect stderr
+    to stdout, we port commands.getstatusoutput to make code compatible
+    Return (status, output) of executing cmd in a shell.
+
+    :param cmd: Command line to run.
+    :type cmd: str
+
+    :return: Exit status and command output(stdout or stderr).
+    :rtype: tuple
+    """
+    cmd += ' 2>&1'
+    result = run(cmd, shell=True, ignore_status=True, verbose=False)
+    text = result.stdout
+    sts = result.exit_status
+    if text[-1:] == '\n':
+        text = text[:-1]
+    return sts, text
```

Comment:
We have no such function returning this exact information - (exit_code, combined_output), but we do have that information, meaning you can use what we already have in place:
```python
>>> print(process.run('ls .git foo', allow_output_check='combined', ignore_status=True).exit_status)
2

>>> print(process.run('ls .git foo', allow_output_check='combined', ignore_status=True).stdout)
ls: cannot access 'foo': No such file or directory
.git:
branches
COMMIT_EDITMSG
config
FETCH_HEAD
GIT_COLA_MSG
HEAD
hooks
index
info
logs
objects
ORIG_HEAD
packed-refs
refs
``` 

Would that be enough? If not, you can use the `allow_output_check='combined'` here instead of `cmd += ' 2>&1'` and also accept all the others `run()` arguments in `getstatusoutput()`, setting your defaults accordingly  (`timeout=None`, `verbose=False`, `ignore_status=True`, `allow_output_check='combined'`, `shell=True`, ...).